WordPress/src/main/java/org/wordpress/android/ui/qrcodeauth/QRCodeAuthViewModel.kt

```diff
@@ -174,14 +206,23 @@ class QRCodeAuthViewModel @Inject constructor(
             return
         }
 
-        clearProperties()
-        // todo: authStore.authenticate call and remove below
         viewModelScope.launch {
-            delay(2000L)
-            postUiState(uiStateMapper.mapDone(::dismissClicked))
+            val result = authStore.authenticate(data = data, token = token)
+            if (result.isError) {
+                postUiState(mapScanErrorToErrorState(result.error))
```

Comment:
Not related to this PR, can we recheck if 

```
private fun postUiState(state: QRCodeAuthUiState) {
    viewModelScope.launch {
        _uiState.value = state
    }
}
```
can be replaced with 

```
private fun postUiState(state: QRCodeAuthUiState) {
    _uiState.value = state
}
```

Reply:
@ashiagr 🤔  I did have it this was early on and I wasn't getting the results I expected (meaning updates were not posting). I'll make a note in the main issue and retest after all pieces are in the feature branch.

Reply:
Marking it resolved as it is added to the main issue and will be revisited later.